app/src/presetdialog.cpp

```diff
@@ -0,0 +1,100 @@
+#include "presetdialog.h"
+#include "ui_presetdialog.h"
+
+#include <QFile>
+#include <QStandardPaths>
+#include <QDir>
+#include <QSettings>
+
+PresetDialog::PresetDialog(PreferenceManager *preferences, QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::PresetDialog),
+    mPrefs(preferences)
+{
+    ui->setupUi(this);
+
+    initPresets();
+
+    connect(ui->buttonBox, &QDialogButtonBox::accepted, this, &QDialog::accept);
+    connect(ui->buttonBox, &QDialogButtonBox::rejected, this, &QDialog::reject);
```

Comment:
The two buttons have been connected in the `presetdialog.ui`. No need to connect in .cpp again, I will be removing them.